frontend/src/routes/manage/Video/TechnicalDetails.tsx

```diff
@@ -115,35 +117,56 @@ export const TrackInfo: React.FC<TrackInfoProps> = (
         tracks.push({ resolution, mimetype, uri });
     }
 
+    const isSingleStream = flavors.size === 1;
+
     return <section css={className}>
         <h2>{t("manage.my-videos.technical-details.tracks")}</h2>
         <ul css={{ fontSize: 15, marginTop: 8, paddingLeft: 24 }}>
             {Array.from(flavors, ([flavor, tracks]) => <li key={flavor}>
-                {translateFlavors ? flavorTranslation(flavor) : <code>{flavor}</code>}
-                <ul>{tracks
+                {!isSingleStream && (translateFlavors
+                    ? flavorTranslation(flavor)
+                    : <code>{flavor}</code>)
+                }
+                <Tracks {...{ isSingleStream }}>{tracks
                     .sort((a, b) => (a.resolution?.[0] ?? 0) - (b.resolution?.[0] ?? 0))
                     .map((track, i) => <TrackItem key={i} {...track} />)
-                }</ul>
+                }</Tracks>
             </li>)}
```

Comment:
I feel like the `Tracks` component doesn't quite pull its weight. It is just deciding between `<ul>` and `<>`. I think I would rewrite this all a bit. What do you think about this? And removing the `Tracks` component?

```tsx
{Array.from(flavors, ([flavor, tracks]) => {
    const trackItems = tracks
        .sort((a, b) => (a.resolution?.[0] ?? 0) - (b.resolution?.[0] ?? 0))
        .map((track, i) => <TrackItem key={i} {...track} />);
    const flavorLabel = translateFlavors 
        ? flavorTranslation(flavor) 
        : <code>{flavor}</code>;
    const flat = isSingleStream && translateFlavors;

    return (
        <li key={flavor}>
            {flat ? trackItems : <>{flavorLabel}{trackItems}</>}
        </li>
    );
}}
```

This also has a behavior change: it only shows the "flat" style (with no nested list and the label removed) when translations are requested, i.e. only on the video page. The technical details page will keep the explicit nesting with flavor label. Which I sliiightly prefer.

Reply:
This also removes the nesting from the technical details page and dual streams, which I think look off without. I adjusted your suggestion to keep that.